internal/lsp/configuration.go

```diff
@@ -22,3 +23,31 @@ func (s *Server) Config(section string) interface{} {
 	}
 	return v
 }
+
+var regFormatCapability = protocol.RegistrationParams{
+	Registrations: []protocol.Registration{
+		{ID: "FormCap",
+			Method:          "textDocument/formatting",
+			RegisterOptions: protocol.TextDocumentRegistrationOptions{DocumentSelector: []protocol.DocumentFilter{{Language: "ttcn3"}}},
+		}}}
+var unregFormatCapability = protocol.UnregistrationParams{
+	Unregisterations: []protocol.Unregistration{{ID: "FormCap", Method: "textDocument/formatting"}},
+}
+
+func (s *Server) didChangeConfiguration(ctx context.Context, _ *protocol.DidChangeConfigurationParams) error {
+	if format, ok := s.Config("ttcn3.server.useOwnFormatter").(bool); ok {
+
+		if format == s.serverConfig.UseOwnFormatter {
+			return nil
+		}
+		if s.serverConfig.UseOwnFormatter {
+			log.Debug("didChangeConfiguration: unregister formatter capability\n")
+			s.client.UnregisterCapability(ctx, &unregFormatCapability)
+		} else {
+			log.Debug("didChangeConfiguration: register formatter capability\n")
+			s.client.RegisterCapability(ctx, &regFormatCapability)
```

Comment:
I think (un)regFormatCapability does not need to be a global variable. We probably can inline it?